packages/core/src/2d/text/TextRenderer.ts

```diff
@@ -438,30 +424,34 @@ export class TextRenderer extends Renderer {
     }
   }
 
-  private _updatePosition(): void {
-    const localPositions = this._sprite._positions;
-    const localVertexPos = TextRenderer._tempVec3;
-    const worldMatrix = this.entity.transform.worldMatrix;
-
-    const { _positions } = this;
-    for (let i = 0, n = _positions.length; i < n; i++) {
-      const curVertexPos = localPositions[i];
-      localVertexPos.set(curVertexPos.x, curVertexPos.y, 0);
-      Vector3.transformToVec3(localVertexPos, worldMatrix, _positions[i]);
+  private _resetCharFont(): void {
+    const lastCharFont = this._charFont;
+    if (lastCharFont) {
+      lastCharFont._addRefCount(-1);
+      lastCharFont.destroy();
     }
+    this._charFont = Font.createFromOS(
+      this.engine,
+      TextUtils.getNativeFontHash(this.font.name, this.fontSize, this.fontStyle)
+    );
+    this._charFont._addRefCount(1);
   }
 
-  private _clearTexture(): void {
-    const { _sprite } = this;
-    // Remove sprite from dynamic atlas.
-    this.engine._dynamicTextAtlasManager.removeSprite(_sprite);
-    this.shaderData.setTexture(SpriteRenderer._textureProperty, null);
-    _sprite.atlasRegion = _sprite.region;
+  private _updatePosition() {
+    const worldMatrix = this.entity.transform.worldMatrix;
+    const { _charRenderDatas } = this;
+    for (let i = 0, n = _charRenderDatas.length; i < n; ++i) {
+      const { localPositions, renderData } = _charRenderDatas[i];
+      for (let j = 0; j < 4; ++j) {
+        Vector3.transformToVec3(localPositions[j], worldMatrix, renderData.positions[j]);
```

Comment:
I am considering an optimization scheme. Only transform one point to world and all position equal this point+offet.

This maybe can improve a lot of perfromance.